desktop/core/src/desktop/js/apps/storageBrowser/InputModal/InputModal.tsx

```diff
@@ -0,0 +1,81 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import React, { useRef } from 'react';
+import Modal from 'cuix/dist/components/Modal';
+import { Input, InputRef } from 'antd';
+import { i18nReact } from '../../../utils/i18nReact';
+
+import './InputModal.scss';
+
+interface InputModalProps {
+  cancelText?: string;
+  description: string;
+  inputLabel: string;
+  inputPlaceholder?: string;
+  okText?: string;
+  onCancel: () => void;
+  onOk: (name: string) => void;
+  showModal: boolean;
+  title: string;
+}
+
+const defaultProps = {
+  cancelText: 'Cancel',
+  inputPlaceholder: 'Enter text here',
+  okText: 'Submit'
+};
+
+const InputModal: React.FC<InputModalProps> = ({
+  cancelText,
+  description,
+  inputLabel,
+  inputPlaceholder,
+  okText,
+  onCancel,
+  onOk,
+  showModal,
+  title
+}): JSX.Element => {
+  const { t } = i18nReact.useTranslation();
+  const textInput = useRef<InputRef>(null);
+
+  const extractText = (): string => {
```

Comment:
I would use the onChange of the Input event to update a state variable. Then you could also use the state variable to enable/disable the Create button, so that it is disabled when there the input is empty.

Remember to clear that state on  onCancel & onOk